contracts/DCNTToken.sol

```diff
@@ -2,46 +2,45 @@
 pragma solidity ^0.8.19;
 
 import { ERC20, ERC20Votes, ERC20Permit } from "@openzeppelin/contracts/token/ERC20/extensions/ERC20Votes.sol";
-import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { AccessControl } from "@openzeppelin/contracts/access/AccessControl.sol";
+import { IDCNTMintAuthorization } from "./mint/IDCNTMintAuthorization.sol";
 
 // @todo handle initial minting correctly. Can it be done inside the constructor?
 
 /// @notice the dcnt token
-contract DCNTToken is ERC20Votes, Ownable {
-    uint128 public nextMint; // Timestamp
-    uint32 public constant MINIMUM_MINT_INTERVAL = 365 days;
-    uint8 public constant MINT_CAP_BPS = 200; // 2%
-
-    error MintExceedsMaximum();
-    error MintTooSoon();
-
-    /// @param _supply amount of tokens to mint at Token Generation Event
-    constructor(uint256 _supply, address _owner) ERC20("Decent", "DCNT") ERC20Permit("Decent") {
+contract DCNTToken is ERC20Votes, AccessControl {
+    IDCNTMintAuthorization public mintAuthorization;
+    bytes32 public constant MINT_ROLE = keccak256("MINT_ROLE");
+    bytes32 public constant UPDATE_MINT_AUTHORIZATION_ROLE = keccak256("UPDATE_MINT_AUTHORIZATION_ROLE");
+    error MintAuthorization();
+
+    constructor(uint256 _supply, address _owner, IDCNTMintAuthorization _mintAuthorization) ERC20("Decent", "DCNT") ERC20Permit("Decent") {
+        _grantRole(MINT_ROLE, _owner);
+        _grantRole(UPDATE_MINT_AUTHORIZATION_ROLE, _owner);
         _mint(msg.sender, _supply);
-        nextMint = uint128(block.timestamp + MINIMUM_MINT_INTERVAL);
-        _transferOwnership(_owner);
+        mintAuthorization = _mintAuthorization;
     }
 
-    /// @notice mint can be called at most once every 365 days,
-    ///  and with an amount no more than 2% of the current supply
+    /// @notice public function to be used for minting new tokens
     /// @param dest address to assign newly minted tokens to
     /// @param amount amount of tokens to mint
-    /// @dev only the `owner` is authorized to mint more tokens
-    function mint(address dest, uint256 amount) external onlyOwner {
-        if (amount > (totalSupply() * MINT_CAP_BPS) / 10000) {
-            revert MintExceedsMaximum();
-        }
-
-        if (block.timestamp < nextMint) {
-            revert MintTooSoon();
+    /// @dev only accounts with `MINT_ROLE` (the DAO) are authorized to mint more tokens
+    function mint(address dest, uint256 amount) external onlyRole(MINT_ROLE) {
+        if (!mintAuthorization.authorizeMint(dest, amount)) {
+            revert MintAuthorization();
         }
-
-        nextMint = uint128(block.timestamp + MINIMUM_MINT_INTERVAL);
         _mint(dest, amount);
     }
 
-    /// @dev holders can burn their own tokens
+    /// @notice holders can burn their own tokens
     function burn(uint256 amount) external {
```

Comment:
I saw that you removed this in the next PR, but maybe it would be good to allow the owner to burn their own tokens, via the same `MINT_ROLE`.

Having the ability to mint new tokens but no deflationary option seems like it might bite us, even though you can always just send them to a burn address, it's cleaner to have a burn option, for as long as we can mint.

Reply:
> allow the owner to burn their own tokens

To be clear, by owner do you mean the DAO (and only the DAO) to burn its own tokens?

Reply:
> 

yeah, via the `MINT_ROLE`

Reply:
I like that idea